tests/e2e/workloads/pvc_snapshot_and_clone/test_compressed_sc_and_support_snap_clone.py

```diff
@@ -141,3 +99,129 @@ def test_compressed_sc_and_support_snap_clone(
+        )
+
+        # Deploy PGSQL workload
+        log.info("Deploying pgsql workloads")
+        pgsql = pgsql_factory_fixture(replicas=1, sc_name=sc_obj.name)
+
+        # Get postgres pvc list obj
+        postgres_pvcs_obj = pgsql.get_postgres_pvc()
+
+        # Create snapshot and clone
+        self.create_snapshot_and_clone(
+            pgsql,
+            postgres_pvcs_obj,
+            sc_obj.name,
+            multi_snapshot_factory,
+            multi_snapshot_restore_factory,
+            multi_pvc_clone_factory,
+        )
+
+    @skipif_ocs_version("<4.8")
+    @skipif_ocp_version("<4.8")
+    @pytest.mark.parametrize(
+        argnames=["kv_version", "replica", "compression"],
+        argvalues=[
+            pytest.param(
+                "v1", 3, "aggressive", marks=pytest.mark.polarion_id("OCS-2707")
+            ),
+            pytest.param(
+                "v2", 3, "aggressive", marks=pytest.mark.polarion_id("OCS-2712")
+            ),
+        ],
+    )
+    def test_encrypted_compressed_sc_and_support_snap_clone(
+        self,
+        kv_version,
+        replica,
+        compression,
+        pv_encryption_kms_setup_factory,
+        storageclass_factory,
+        pgsql_factory_fixture,
+        multi_snapshot_factory,
+        multi_snapshot_restore_factory,
+        multi_pvc_clone_factory,
+        pgsql_teardown,
+    ):
+        """
+        1. Create new sc with compression and encryption enabled
+        2. Deploy PGSQL workload using those new sc created
+        3. Take a snapshot of the pgsql PVC.
+        4. Create a new PVC out of that snapshot or restore snapshot
+        5. Attach a new pgsql pod to it.
+        6. Resize the new PVC
+        7. Clone pgsql PVC and attach a new pgsql pod to it
+        8. Resize cloned PVC
+        """
+
+        log.info("Setting up csi-kms-connection-details configmap")
+        self.vault = pv_encryption_kms_setup_factory(kv_version)
+        log.info("csi-kms-connection-details setup successful")
+
+        # Create an encryption enabled storageclass for RBD
+        sc_obj = storageclass_factory(
+            interface=CEPHBLOCKPOOL,
+            encrypted=True,
+            encryption_kms_id=self.vault.kmsid,
+        )
+
+        # Create ceph-csi-kms-token in the tenant namespace
+        self.vault.vault_path_token = self.vault.generate_vault_token()
+        self.vault.create_vault_csi_kms_token(namespace=BMO_NAME)
+
+        # Deploy PGSQL workload
+        log.info("Deploying pgsql workloads")
+        pgsql = pgsql_factory_fixture(replicas=1, sc_name=sc_obj.name)
+
+        # Get postgres pvc list obj
+        postgres_pvcs_obj = pgsql.get_postgres_pvc()
+
+        # Create snapshot and clone
+        self.create_snapshot_and_clone(
+            pgsql,
+            postgres_pvcs_obj,
+            sc_obj.name,
+            multi_snapshot_factory,
+            multi_snapshot_restore_factory,
+            multi_pvc_clone_factory,
+        )
```

Comment:
Right. It would be better to add validation in those snapshots, restore and clone fixtures than adding in each test for key deletion from the Vault side. 
Will take this another PR, wdyt?

Reply:
Ack